### DISCUSSION TITLE
Licensing new code containing xarray

### DISCUSSION THREAD

**Discussion by catemgn** [pydata/xarray#6264] [Q&A] [answered]
Hi!

I have been using a lot xarray for my PhD, and now I would like now release the code of some piece of work. I used xarray library and functionalities 'as-is', without modifications in the source code. 

Which licenses I can release my code under given that xarray is distributed under Apache 2.0 license? Would MIT work?

I am new to the topic and I tried to navigate the QA online about licensing, but I didn't come out with a satisfactory answer. So I am asking to those directly involved!

Thanks in advance! 🙏 

Caterina

**Comment by ianliu** [comment]
I think your use case falls into the category of *Citing Xarray*, as stated here:

https://docs.xarray.dev/en/stable/getting-started-guide/faq.html#how-should-i-cite-xarray

**Comment by keewis** [comment, marked as answer]
What we have been doing for packages that depend on `xarray` is this:
- if the package only depends on `xarray` (it is installed into the environment), there is no restriction on its license
- if the package copies (code from) `xarray` ("vendoring"), the license of the package itself is still up to the author, but the vendored code must be marked as such (a comment should be enough) and the license of `xarray` should be included somewhere, usually a `licenses` directory.
- if the copied code was modified: same as if it was copied, but the comment should state that it was modified

**Reply by catemgn** [reply]
yes I think my case fall in 'the package only depends on xarray'. Thanks for your complete reply!
